src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,229 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useLayoutEffect, useState, createRef, useEffect, useCallback, useMemo } from 'react';
+import { TableProps } from './interfaces';
+import clsx from 'clsx';
+interface CellOffsets {
+  first: number[];
+  last: number[];
+}
+interface StickyStyles {
+  sticky: {
+    left?: string;
+    right?: string;
+  };
+  stuck: {
+    paddingLeft?: string;
+  };
+}
+
+interface StickyColumnParams {
+  containerWidth: number | null;
+  hasSelection: boolean;
+  isWrapperScrollable: boolean;
+  stickyColumns?: TableProps.StickyColumns;
+  visibleColumnsLength: number;
+  tableRefObject: React.RefObject<HTMLTableElement>;
+  wrapperRefObject: React.RefObject<HTMLDivElement>;
+}
+
+export interface GetStickyColumnProperties {
+  isSticky: boolean;
+  isLastStickyLeft: boolean;
+  isLastStickyRight: boolean;
+  stickyStyles: StickyStyles;
+}
+
+// We allow the table to have a minimum of 148px of available space besides the sum of the widths of the sticky columns
+const MINIMUM_SCROLLABLE_SPACE = 148;
+
+export const useCellOffsets = (tableCellRefs: Array<React.RefObject<HTMLTableCellElement>>) => {
```

Comment:
If not used outside this file, better not to export it. This saves time on review and future refactoring, because people can understand that this is a local function

Reply:
Nice catch. Leftovers from refactoring. Will re-check the changes to make sure there are no more useless exports.